src/contracts/misc/RwaATokenManager.sol

```diff
@@ -13,46 +15,63 @@ import {IRwaATokenManager} from '../interfaces/IRwaATokenManager.sol';
  */
 contract RwaATokenManager is AccessControl, IRwaATokenManager {
   /// @inheritdoc IRwaATokenManager
-  bytes32 public constant override AUTHORIZED_TRANSFER_ROLE = keccak256('AUTHORIZED_TRANSFER');
+  IPool public immutable POOL;
+
+  /// @inheritdoc IRwaATokenManager
+  bytes32 public constant AUTHORIZED_TRANSFER_ROLE = keccak256('AUTHORIZED_TRANSFER');
 
   /**
    * @dev Constructor
    * @param owner The address of the default admin role
+   * @param addressesProvider The address of the PoolAddressesProvider of Aave V3 Pool
    */
-  constructor(address owner) {
+  constructor(address owner, address addressesProvider) {
     _setupRole(DEFAULT_ADMIN_ROLE, owner);
+    POOL = IPool(IPoolAddressesProvider(addressesProvider).getPool());
   }
 
   /// @inheritdoc IRwaATokenManager
-  function grantAuthorizedTransferRole(address aTokenAddress, address account) external override {
-    grantRole(getAuthorizedTransferRole(aTokenAddress), account);
+  function grantAuthorizedTransferRole(address asset, address account) external override {
+    grantRole(getAuthorizedTransferRole(asset), account);
   }
 
   /// @inheritdoc IRwaATokenManager
-  function revokeAuthorizedTransferRole(address aTokenAddress, address account) external override {
-    revokeRole(getAuthorizedTransferRole(aTokenAddress), account);
+  function revokeAuthorizedTransferRole(address asset, address account) external override {
+    revokeRole(getAuthorizedTransferRole(asset), account);
```

Comment:
@miguelmtzinf one possible issue here is that if you are going to drop the reserve in the `Pool` **before** calling `revokeAuthorizedTransferRole`, the `revokeAuthorizedTransferRole` will revert because `POOL.getReserveAToken(asset)` will return `address(0)`

Reply:
It's not going to be an issue I think because you can drop the reserve only if there's no more suppliers (see `ValidationLogic.validateDropReserve`). This mean that you have no more `RwaAToken` to force remove, but still, I have to underlying this.